docs/API.md

```diff
@@ -81,151 +119,394 @@ Jest uses Jasmine 2 by default. An introduction to Jasmine 2 can be found
 
-### `jest.runAllTimers()`
-Exhausts the **macro**-task queue (i.e., all tasks queued by `setTimeout()` and `setInterval()`).
+```js
+describe('beverage registration', () => {
+  it('applies correctly to orange La Croix', () => {
+    let beverage = new LaCroix('orange');
+    register(beverage);
+    let f = jest.fn();
+    applyToAll(f);
+    expect(f).toBeCalledWith(beverage);
+  });
+});
+```
 
-When this API is called, all pending "macro-tasks" that have been queued via `setTimeout()` or `setInterval()` will be executed. Additionally if those macro-tasks themselves schedule new macro-tasks, those will be continually exhausted until there are no more macro-tasks remaining in the queue.
+### `.toBeCloseTo(number, delta)`
 
-This is often useful for synchronously executing setTimeouts during a test in order to synchronously assert about some behavior that would only happen after the `setTimeout()` or `setInterval()` callbacks executed. See the [Timer mocks](/jest/docs/timer-mocks.html) doc for more information.
+Using exact equality with floating point numbers is a bad idea. Rounding means that intuitive things fail. For example, this test fails:
 
-### `jest.runOnlyPendingTimers()`
-Executes only the macro-tasks that are currently pending (i.e., only the tasks that have been queued by `setTimeout()` or `setInterval()` up to this point). If any of the currently pending macro-tasks schedule new macro-tasks, those new tasks will not be executed by this call.
+```js
+describe('adding numbers', () => {
+  it('works sanely with simple decimals', () => {
+    expect(0.2 + 0.1).toBe(0.3); // Fails!
+  });
+});
+```
 
-This is useful for scenarios such as one where the module being tested schedules a `setTimeout()` whose callback schedules another `setTimeout()` recursively (meaning the scheduling never stops). In these scenarios, it's useful to be able to run forward in time by a single step at a time.
+It fails because in JavaScript, `0.2 + 0.1` is actually `0.30000000000000004`. Sorry.
 
-### `jest.setMock(moduleName, moduleExports)`
-Explicitly supplies the mock object that the module system should return for the specified module.
+Instead, use `.toBeCloseTo` with a small delta. A valid test for this same addition problem is:
 
-On occasion there are times where the automatically generated mock the module system would normally provide you isn't adequate enough for your testing needs. Normally under those circumstances you should write a [manual mock](/jest/docs/manual-mocks.html) that is more adequate for the module in question. However, on extremely rare occasions, even a manual mock isn't suitable for your purposes and you need to build the mock yourself inside your test.
+```js
+describe('adding numbers', () => {
+  it('works sanely with simple decimals', () => {
+    expect(0.2 + 0.1).toBeCloseTo(0.3, 0.000000001);
```

Comment:
@valerybugakov i remember that there was something tricky about this matcher. Can you look at it, since you implemented it :)